src/coreclr/dlls/mscorrc/CMakeLists.txt

```diff
@@ -3,20 +3,9 @@ include_directories("../../pal/prebuilt/corerror")
 add_definitions(-DFX_VER_INTERNALNAME_STR=mscorrc.dll)
 
 if(CLR_CMAKE_HOST_WIN32)
-    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /NOENTRY")
-
-    # remove /guard:cf, /guard:ehcont, and /CETCOMPAT from resource-only libraries
-    set_property(DIRECTORY PROPERTY CLR_CONTROL_FLOW_GUARD OFF)
-
-    if (CLR_CMAKE_HOST_ARCH_AMD64)
-        set_property(DIRECTORY PROPERTY CLR_EH_CONTINUATION OFF)
-        string(REPLACE "/CETCOMPAT" "" CMAKE_SHARED_LINKER_FLAGS ${CMAKE_SHARED_LINKER_FLAGS})
-    endif (CLR_CMAKE_HOST_ARCH_AMD64)
-
-    add_library_clr(mscorrc SHARED
+    add_library_clr(mscorrc OBJECT
         include.rc
```

Comment:
Could you please delete `#include "....dlls/mscorrc/resource.h"` from as many places as possible (in particular the ones under debug\di) to make sure that these resources strings are not used by non-coreclr.dll binaries?